Don't escape backslash in patterns to allow backslash in patterns

vds2212 opened the pull request:
To solve #346 and allow backslash in pattern searching.

vds2212 committed:
don't escape backslash in patterns to allow backslash in patterns

dyng commented:
I think escaping itself is required for building a valid command, but for your issue you could find out which escaping is right. (especially on Windows)

Please refer to below issue, it seems related to your problem:

https://github.com/dyng/ctrlsf.vim/issues/164

vds2212 commented:
Thanks!

I'll have a second look and propose a solution that work for both: my use case and the error reported.

vds2212 committed:
On Linux "\\" leads to \ being received by the process
On Windows "\\" lead to \\ being received by the process
The string need to be escaped differently:
On Linux both \ and " need to be escaped
On Windows only " need to be escaped with another " ("")

dyng commented:
Merged. Thanks for your contribution!
